### PR TITLE
Added refresh token flag

### LINKED ISSUE #13: load a refresh token to log in
drg login --token <refresh token> alternative … in case the browser flow doesn't work

### PR DESCRIPTION
**Pull request by vedangj044**
Fixes #13 

Added a `--token` flag to the login user flow, 
Works like

`drg login <login_url> --token $(cat token.txt)`

### CODE REVIEW

File: src/openid.rs [hunk 1]
```diff
@@ -48,6 +52,30 @@ pub fn login(api_endpoint: Url) -> Result<Context> {
...
+        Some(token_url.clone()),
+    );
+
+    // Exchange the refresh token for access token
+    client
+        .exchange_refresh_token(&oauth2::RefreshToken::new(refresh_token_val.to_string()))
+        .request(http_client)
+        .map_err(|_| Error::msg("Invalid refresh token"))
+}
+
```
Comment (jbtrystram) [inline src/openid.rs:78]: We would have two `refresh_token` methods that would do almost the same thing. 
I think you could refactor both to use the same call to `client.exchange_refresh_token()`. 

https://github.com/drogue-iot/drg/blob/3acc5863a734247bb6df263f272a2f5d4ab257b2/src/openid.rs#L126-L155

**Review by jbtrystram — CHANGES_REQUESTED** [review]

File: src/openid.rs [hunk 2]
```diff
@@ -22,13 +22,17 @@ const CLIENT_ID: &str = "drogue";
...
-pub fn login(api_endpoint: Url) -> Result<Context> {
+pub fn login(api_endpoint: Url, refresh_token_val: Option<&str>) -> Result<Context> {
     log::info!("Starting authentication process with {}", api_endpoint);
 
     let (sso_url, registry_url) = util::get_drogue_services_endpoint(api_endpoint.clone())?;
     let (auth_url, token_url) = util::get_auth_and_tokens_endpoints(sso_url)?;
 
-    let token = get_token(auth_url.clone(), token_url.clone())?;
+    let token = match refresh_token_val {
+        Some(i) => login_with_refresh_token(auth_url.clone(), token_url.clone(), i)?,
```
[the remark this answers is not in the document]
Comment (vedangj044) [reply src/openid.rs:32]: Sure, let's keep it `refresh_token_var`

File: src/openid.rs (hunk 1 above)
Comment (vedangj044) [reply src/openid.rs:78]: Okay good advice, I have moved the repeated code to a new function `exchange_token`

**Comment by jbtrystram** [comment]
Make sure to document the feature in the README as well

**Review by jbtrystram — APPROVED** [review]
LGTM

**Pull request merged**